src/utils.js

```diff
@@ -0,0 +1,144 @@
+const core = require('@actions/core');
+const fs = require('fs');
+const glob = require('glob');
+const xmlParser = require('fast-xml-parser');
+const he = require('he');
+
+function findTrxFiles(baseDir) {
+  core.info(`Looking for trx files in '${baseDir}'...`);
+
+  const files = glob.sync(baseDir + '/**/*.trx', {});
+  if (!files || files.length === 0) {
+    throw new RangeError('There were no trx files found.');
+  }
+
+  core.info(`The following trx files were found:`);
+  core.info(`\t${files.join('\n\t')}`);
+  return files;
+}
+
+async function transformAllTrxToJson(trxFiles) {
+  const transformedTrxReports = [];
+
+  for (const trx of trxFiles) {
+    transformedTrxReports.push(await transformTrxToJson(trx));
+  }
+
+  return transformedTrxReports;
+}
+
+async function transformTrxToJson(filePath) {
+  let trxDataWrapper;
+
+  core.info(`Transforming file ${filePath}`);
+
+  const xmlData = fs.readFileSync(filePath, 'utf-8');
+  const options = {
+    attributeNamePrefix: '_',
+    textNodeName: '#text',
+    ignoreAttributes: false,
+    ignoreNameSpace: false,
+    allowBooleanAttributes: true,
+    parseNodeValue: true,
+    parseAttributeValue: true,
+    trimValues: true,
+    cdataTagName: '__cdata',
+    cdataPositionChar: '\\c',
+    parseTrueNumberOnly: false,
+    arrayMode: false,
+    attrValueProcessor: (val, _attrName) =>
+      he.decode(val, {
+        isAttributeValue: true
+      }),
+    tagValueProcessor: (val, _tagName) => he.decode(val),
+    stopNodes: ['parse-me-as-string']
+  };
+
+  if (xmlParser.validate(xmlData.toString()) === true) {
+    const jsonString = xmlParser.parse(xmlData, options, true);
```

Comment:
From reading their documentation, [passing `true` as that last argument triggers validation](https://github.com/NaturalIntelligence/fast-xml-parser#xml-to-json) and will throw an error if it fails. It sort of seems redundant to do that when we validate in the `if` check, but it's not a big deal I guess.

Reply:
I'm not sure if it produces different results when it's done with different calls.  In this case I think I'd rather leave it since I don't know what it's doing.